homeassistant/components/notion/__init__.py

```diff
@@ -0,0 +1,311 @@
+        await hass.config_entries.async_forward_entry_unload(
+            config_entry, component)
+
+    return True
+
+
+async def register_new_bridge(hass, bridge, config_entry_id):
+    """Register a new bridge."""
+    device_registry = await dr.async_get_registry(hass)
+    device_registry.async_get_or_create(
+        config_entry_id=config_entry_id,
+        identifiers={
+            (DOMAIN, bridge['hardware_id'])
+        },
+        manufacturer='Silicon Labs',
+        model=bridge['hardware_revision'],
+        name=bridge['name'] or bridge['id'],
+        sw_version=bridge['firmware_version']['wifi']
+    )
+
+
+class Notion:
+    """Define a class to handle the Notion API."""
+
+    def __init__(self, hass, client, config_entry_id):
+        """Initialize."""
+        self._client = client
+        self._config_entry_id = config_entry_id
+        self._hass = hass
+        self.bridges = {}
+        self.sensors = {}
+        self.tasks = {}
+
+    async def async_update(self):
+        """Get the latest Notion data."""
+        from aionotion.errors import NotionError
+
+        tasks = {
+            'bridges': self._client.bridge.async_all(),
+            'sensors': self._client.sensor.async_all(),
+            'tasks': self._client.task.async_all(),
+        }
+
+        results = await asyncio.gather(*tasks.values(), return_exceptions=True)
+        for attr, result in zip(tasks, results):
+            if isinstance(result, NotionError):
+                _LOGGER.error(
+                    'There was an error while updating %s: %s', attr, result)
+                continue
+
+            holding_pen = getattr(self, attr)
+            for item in result:
+                if attr == 'bridges' and item['id'] not in holding_pen:
+                    # If a new bridge is discovered, register it:
+                    self._hass.async_create_task(
+                        register_new_bridge(
+                            self._hass, item, self._config_entry_id))
+                holding_pen[item['id']] = item
+
+
+class NotionEntity(Entity):
+    """Define a base Notion entity."""
+
+    def __init__(
+            self,
+            notion,
+            task_id,
+            sensor_id,
+            bridge_id,
+            system_id,
+            name,
+            device_class):
+        """Initialize the entity."""
+        self._async_unsub_dispatcher_connect = None
+        self._attrs = {ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION}
+        self._bridge_id = bridge_id
+        self._device_class = device_class
+        self._name = name
+        self._notion = notion
+        self._sensor_id = sensor_id
+        self._state = None
+        self._system_id = system_id
+        self._task_id = task_id
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return any(
+            self._task_id == task_id for task_id in list(self._notion.tasks))
```

Comment:
Since tasks is a dictionary and `list(dictionary)` returns an array of keys, you can just test `self._task_id in self._notion.tasks`